# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1215619791 -7200
#      Wed Jul 09 18:09:51 2008 +0200
# Node ID 7ba62ea0f1618e7659d4ff864ef9c09489f9b9be
# Parent  47a18b8c9000d54f3778479b03e44d98b2762ec2
Fix seg-fault for dbstep at end of function

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -24,16 +24,22 @@ 2008-07-09  Michael Goffioul  <michael.g
 	vertex_data::vertex_data(void), vertex_data::vertex_data(...)):
 	initialize reference counting correctly.
 
 2008-07-09  John W. Eaton  <jwe@octave.org>
 
 	* toplev.cc (main_loop): Unwind-protect global_command.
 	* input.cc (get_debug_input): Likewise.
 
+2008-07-09  David Bateman  <dbateman@free.fr>
+
+	* pt-id.cc (octave_value_list tree_identifier::rvalue (int), 
+	octave_lvalue tree_identifier::lvalue (void)): Remove 
+	MAYBE_DO_BREAKPOINT.
+
 2008-07-08  John W. Eaton  <jwe@octave.org>
 
 	* graphics.h.in (axes::properties::keypos): Declare as
 	any_property instead of double_property.
 
 2008-06-28  John W. Eaton  <jwe@octave.org>
 
 	* debug.cc (push_dummy_call_stack_elt): Delete.
diff --git a/src/pt-id.cc b/src/pt-id.cc
--- a/src/pt-id.cc
+++ b/src/pt-id.cc
@@ -52,18 +52,16 @@ tree_identifier::eval_undefined_error (v
 	     name ().c_str (), l, c);
 }
 
 octave_value_list
 tree_identifier::rvalue (int nargout)
 {
   octave_value_list retval;
 
-  MAYBE_DO_BREAKPOINT;
-
   if (error_state)
     return retval;
 
   octave_value_list evaluated_args;
   bool args_evaluated = false;
 
   octave_value val = xsym().find (0, string_vector (), evaluated_args,
 				  args_evaluated);
@@ -112,18 +110,16 @@ tree_identifier::rvalue (void)
     retval = tmp(0);
 
   return retval;
 }
 
 octave_lvalue
 tree_identifier::lvalue (void)
 {
-  MAYBE_DO_BREAKPOINT;
-
   return octave_lvalue (&(xsym().varref ()));
 }
 
 tree_identifier *
 tree_identifier::dup (symbol_table::scope_id sc,
 		      symbol_table::context_id context)
 {
   // The new tree_identifier object contains a symbol_record
